src/pkg/docs/service/directory/directory.go

```diff
@@ -81,20 +250,102 @@ func (s *Service) Delete(ctx context.Context, req processing.RequestInterface, s
 	return objectVersionID.String(), nil
 }
 
-// TODO
-func (s *Service) GetByTime(ctx context.Context, systemID string, ehrUUID *uuid.UUID, userID string, versionTime time.Time) (*model.Directory, error) {
-	return nil, errors.ErrNotImplemented
+func (s *Service) GetByTime(ctx context.Context, systemID string, ehrUUID *uuid.UUID, patientID string, versionTime time.Time) (*model.Directory, error) {
+	docMeta, err := s.Infra.Index.GetDocByTime(ctx, ehrUUID, types.Directory, uint32(versionTime.Unix()))
+
+	if err != nil {
+		if errors.Is(err, errors.ErrNotFound) {
+			docMeta, err = s.Infra.Index.GetDocLastByType(ctx, ehrUUID, types.Directory)
+			if err != nil {
+				return nil, fmt.Errorf("Index.GetDocLastByType error: %w", err)
+			}
+		} else {
+			return nil, fmt.Errorf("Index.GetDocByTime error: %w ehrID %s nearestTime %s docType %s", err, ehrUUID.String(), versionTime.String(), types.Directory)
+		}
+	}
```

Comment:
I think, we can create function `tryGetDocMet`

Reply:
I don't think you need to ask for the latest version of the document here. 
If there is no document for the specified date, we return 404

Reply:
specifications have got rule: "retrieves the version extant at specified time, otherwise retrieves the latest directory FOLDER version. "